pkg/service/auth_context.go

```diff
@@ -18,6 +18,29 @@ var (
 	authCtxLog = ctrl.Log.WithName("Authorino").WithName("AuthContext")
 )
 
+type EvaluationResponse struct {
+	Evaluator common.AuthConfigEvaluator
+	Success   bool
+	Object    interface{}
+	Error     error
+}
+
+func newEvaluationResponseSuccess(evaluator common.AuthConfigEvaluator, obj interface{}) EvaluationResponse {
+	return EvaluationResponse{
+		Evaluator: evaluator,
+		Success:   true,
+		Object:    obj,
+	}
+}
+
+func newEvaluationResponseFailure(evaluator common.AuthConfigEvaluator, err error) EvaluationResponse {
+	return EvaluationResponse{
+		Evaluator: evaluator,
+		Success:   false,
+		Error:     err,
+	}
+}
+
```

Comment:
I think here we could simplify  a bit just removing the `Success` from the `EvaluationResponse` struct. Then we could have just one func `newEvaluationResponse` like:
```go
func newEvaluationResponse(evaluator common.AuthConfigEvaluator, obj interface{}, err error) EvaluationResponse {
	return EvaluationResponse{
		Evaluator: evaluator,
                Object: obj,
		Error:     err,
	}
}

```

later it could be called depending on success or failure like:

```go
newEvaluationResponse(evaluator, nil, err) # error found
newEvaluationResponse(evaluator, obj, nil) # as cool as kim deal
```

later on we would just compare the `EvaluationResponse.Error` to `nil` to flow as intended.